cmd/osv-scanner/main.go

```diff
@@ -265,6 +69,53 @@ func run(args []string, stdout, stderr io.Writer) int {
 	return 0
 }
 
+// Gets all valid commands and global options for OSV-Scanner.
+func getAllCommands(commands []*cli.Command) []string {
+	// Adding all subcommands
+	allCommands := make([]string, 0)
+	for _, command := range commands {
+		allCommands = append(allCommands, command.Name)
+	}
+
+	// Adding help command and help flags
+	for _, flag := range cli.HelpFlag.Names() {
+		allCommands = append(allCommands, flag)      // help command
+		allCommands = append(allCommands, "-"+flag)  // help flag
+		allCommands = append(allCommands, "--"+flag) // help flag
+	}
+
+	// Adding version flags
+	for _, flag := range cli.VersionFlag.Names() {
+		allCommands = append(allCommands, "-"+flag)
+		allCommands = append(allCommands, "--"+flag)
+	}
+
+	return allCommands
+}
+
+// Inserts the default command to args if no command is specified.
+func insertDefaultCommand(args []string, stdout, stderr io.Writer, app *cli.App) []string {
+	if len(args) < 2 {
+		return args
+	}
+
+	allCommands := getAllCommands(app.Commands)
+	if !slices.Contains(allCommands, args[1]) {
+		// Avoids modifying args in-place, as some unit tests rely on its original value for multiple calls.
+		argsTmp := make([]string, len(args)+1)
+		copy(argsTmp[2:], args[1:])
+		argsTmp[1] = app.DefaultCommand
+
+		// Executes the cli app with the new args.
+		return argsTmp
+	} else if _, err := os.Stat(args[1]); err == nil {
+		r := reporter.NewJSONReporter(stdout, stderr, reporter.InfoLevel)
+		r.Warnf("Warning: '%v' exists as both a subcommand of OSV-Scanner and as a file in the filesystem. It operates as a command here. If you intend to scan the file, please specify a subcommand.\n", args[1])
```

Comment:
Nit: "It operates as a command here." seems a bit confusing to me. I would reword this to somethin like:

```
Warning: `%s` exists as both a subcommand of OSV-Scanner and as a file on the filesystem. `%s` is assumed to be a subcommand here. If you intended for `%s` to be an argument to `%s`, you must specify `%s %s` in your commandline.
```

Also do we need "%v"? can we use "%s" here?